src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -1188,3 +1169,13 @@ def build_udtf(
         t = ast.kwargs.add()
         t._1 = k
         build_expr_from_python_val(t._2, v)
+
+
+def build_intermediate_stmt(ast_batch: AstBatch, o: Any) -> None:
+    if hasattr(o, "_ast_stmt"):
```

Comment:
can we add docstring and make the name maybe more specific?

Similar to the comment above, can we avoid the `hasattr` pattern here and make things more explicit?